### PR TITLE
feat: profile delete route added

### LINKED ISSUE #49: Delete User/Deactivate account
@route DELETE api/profile/me 
@desc Delete profile, user 
@access Private

### PR DESCRIPTION
**Pull request by Arnab-31**
closes #49

### CODE REVIEW

File: server/src/middleware/auth.js [hunk 1]
```diff
@@ -4,17 +4,17 @@ const keys = require("../config/keys");
 
 const auth = async (req, res, next) => {
   // Get token from the header
-  const token = req.header("x-auth-token");
+  var token = req.header("Authorization");
```
Comment (BhavyaaArora-08) [inline server/src/middleware/auth.js]: Why have you changed this file?

File: server/src/routers/api/profileRouter.js [hunk 2]
```diff
@@ -32,4 +33,28 @@ router.get("/:id", async(req,res)=> {
...
+  try{
+      await UserProfile.deleteOne({userField: req.user._id})
+      await req.user.remove()
+      res.send(req.user);
+  }catch(e){
+      res.status(400).send(e);
+  }
+})
+
+
```
Comment (BhavyaaArora-08) [inline server/src/routers/api/profileRouter.js:59]: Attach screenshots.

File: server/src/middleware/auth.js (hunk 1 above)
Comment (Arnab-31) [reply server/src/middleware/auth.js]: Actually, for some reason postman is not detecting the token if it is not stored in Authorization.

File: server/src/routers/api/profileRouter.js (hunk 2 above)
Comment (Arnab-31) [reply server/src/routers/api/profileRouter.js:59]: image

File: server/src/middleware/auth.js (hunk 1 above)
Comment (BhavyaaArora-08) [reply server/src/middleware/auth.js]: image

Did you set it like this?

File: server/src/routers/api/profileRouter.js (hunk 2 above)
Comment (BhavyaaArora-08) [reply server/src/routers/api/profileRouter.js:59]: In the route your have responded as follows:-
res.send(req.user);

But the screenshot has a completely diff format.

Do the following changes Today itself :- 
1. Undo all the changes made in the auth.js file
2. The response format in the screenshot is correct, make that change in the route.

Comment (Arnab-31) [reply server/src/routers/api/profileRouter.js:59]: @Codess08 I was doing a small mistake in postman. I have fixed the code now. Had to change this line " _id: decoded._id " in the auth file though.

**Pull request merged**